src/test/java/com/github/shoothzj/mjp/client/Mqttv3ClientTest.java

```
@@ -0,0 +1,79 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.github.shoothzj.mjp.client;
+
+import com.github.shoothzj.mjp.MqsarBroker;
+import com.github.shoothzj.mjp.integrate.MqsarTestUtil;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+@Slf4j
+public class Mqttv3ClientTest {
+
+    private MqsarBroker mqsarBroker;
+
+    @Before
+    public void start() throws Exception{
+            mqsarBroker = MqsarTestUtil.setupMqsar();
+    }
+
+    @Test
+    public void connectMqttBroker() throws MqttException {
+        MqttConnectOptions connectOpt = new MqttConnectOptions();
+        connectOpt.setCleanSession(true);
+        connectOpt.setKeepAliveInterval(40);
+        connectOpt.setUserName("admin");
+        connectOpt.setPassword("123456".toCharArray());
+        connectOpt.setConnectionTimeout(10);
+        String url = String.format("tcp://%s:%d", mqsarBroker.getMqttHost(), mqsarBroker.getMqttPort());
+        MqttClient mqttClient = new MqttClient(url, "clientId001");
+        mqttClient.connect(connectOpt);
+        Assert.assertTrue(mqttClient.isConnected());
+    }
+
+    @Test
+    public void connectMqttBrokerFail() {
+        MqttConnectOptions connectOpt = new MqttConnectOptions();
+        connectOpt.setCleanSession(true);
+        connectOpt.setKeepAliveInterval(40);
+        connectOpt.setConnectionTimeout(10);
+        String url = String.format("tcp://%s:%d", mqsarBroker.getMqttHost(), mqsarBroker.getMqttPort());
+        MqttClient mqttClient = null;
+        try {
+            mqttClient = new MqttClient(url, "clientId001");
+            mqttClient.connect(connectOpt);
+        } catch (MqttException e) {
```

we can just throw the exception. It's test case